src/validation.h

```diff
@@ -711,14 +732,170 @@ class SnapshotMetadata
 
 };
 
+/**
+ * Provides an interface for creating and interacting with multiple
+ * chainstates, as well as a means to initialize chainstates from a
+ * UTXO snapshot. Managed chainstates can be maintained at different
+ * heights simultaneously.
+ *
+ * This class provides abstractions that allow the retrieval of the current
+ * most-work chainstate ("Active") as well as chainstates which may be in
+ * background use to validate UTXO snapshots.
+ *
+ * This class was created to displace the use of global variables (e.g.
```

Comment:
In commit "validation: introduce ChainstateManager, use in init" (8d9b707f609b647fb28ae182a097adc03e718725)

Would drop this. Referencing historical variables in commit and review comments is helpful, but confusing in code comments. If this is supposed to suggest that there are still weird things about the code that could be cleaned up in the future, it would probably be better to say directly what the current problems are and how they might be addressed.